compiler/rustc_passes/src/dead.rs

```diff
@@ -37,6 +37,19 @@ fn should_explore(tcx: TyCtxt<'_>, hir_id: hir::HirId) -> bool {
     )
 }
 
+fn base_expr<'a>(expr: &'a hir::Expr<'a>) -> &'a hir::Expr<'a> {
+    let mut current = expr;
+    loop {
+        match current.kind {
+            hir::ExprKind::Field(base, ..) => {
+                current = base;
+            }
+            _ => break,
+        }
+    }
+    current
```

Comment:
I couldn't resist the code golf opportunity

```suggestion
fn base_expr<'a>(mut expr: &'a hir::Expr<'a>) -> &'a hir::Expr<'a>
    loop {
        match expr.kind {
            hir::ExprKind::Field(base, ..) => expr = base,
            _ => return expr,
        }
    }
}
```